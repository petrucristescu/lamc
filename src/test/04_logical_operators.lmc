# Testing operators with ~ prefix syntax
~import operators

~(
    print (false "yes" "no")

    # Test boolean operators
    print (not false "yes" "no")
    print (not true "yes" "no")

    # Test AND operator
    print (and true true "yes" "no")
    print (and true false "yes" "no")
    print (and false true "yes" "no")

    # Test OR operator
    print (or false false "yes" "no")
    print (or true false "yes" "no")
    print (or false true "yes" "no")

    # Test chained boolean operations
    print (not (and true false) "Correctly not-ed" "Should not see this")
    print (or false true "yes" "no")

    # Test complex boolean expression
    print (and (or true false) (not false) "Complex expression works" "Should not see this")
)
