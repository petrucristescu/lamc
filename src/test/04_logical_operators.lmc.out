no
yes
no
yes
no
no
no
yes
yes
Correctly not-ed
yes
Complex expression works
